### PR TITLE
Make the resolveInternalPointer method abide the IAllocator API

### PR DESCRIPTION
**Pull request by edi33416**
Have the resolveInternalPointer method defined in the module's static allocators abide the IAllocator API.

### CODE REVIEW

**Comment by JackStouffer** [comment]
Ping @andralex

File: std/experimental/allocator/building_blocks/bitmapped_block.d [hunk 1]
```diff
@@ -996,22 +996,24 @@ struct BitmappedBlockWithInternalPointers(
     }
 
     /// Ditto
-    void[] resolveInternalPointer(void* p)
+    Ternary resolveInternalPointer(void* p, ref void[] result)
     {
+        result = null;
```
Comment (andralex) [inline std/experimental/allocator/building_blocks/bitmapped_block.d]: The other path ends up assigning result twice, so move this inside the if.

Comment (andralex) [inline std/experimental/allocator/building_blocks/bitmapped_block.d]: On second thought I see you have two more paths that use this, so never mind. Looks good.

Comment (andralex) [inline std/experimental/allocator/building_blocks/bitmapped_block.d]: On third thought: we are under no obligation to assign anything to result if we return anything but `Ternary.yes`, so please remove this line entirely!

File: std/experimental/allocator/building_blocks/null_allocator.d [hunk 2]
```diff
@@ -41,9 +41,10 @@ struct NullAllocator
     /// Returns $(D Ternary.no).
     Ternary owns(void[]) shared const { return Ternary.no; }
     /**
-    Returns $(D null).
+    Returns $(D Ternary.no).
     */
-    void[] resolveInternalPointer(void*) shared const { return null; }
+    Ternary resolveInternalPointer(void*, ref void[]) shared const
+    { return Ternary.no; }
```
Comment (andralex) [inline std/experimental/allocator/building_blocks/null_allocator.d]: not covered, please add unittest

File: std/experimental/allocator/package.d [hunk 3]
```diff
@@ -1930,8 +1930,7 @@ class CAllocatorImpl(Allocator, Flag!"indirect" indirect = No.indirect)
     {
         static if (hasMember!(Allocator, "resolveInternalPointer"))
         {
-            result = impl.resolveInternalPointer(p);
-            return Ternary(result.ptr !is null);
+            return impl.resolveInternalPointer(p, result);
```
Comment (andralex) [inline std/experimental/allocator/package.d]: not covered, please add unittest

**Pull request merged**